tests/candev/Makefile

```diff
@@ -5,7 +5,16 @@ USEMODULE += can
 USEMODULE += isrpipe
 
 # define the CAN driver you want to use here
-CAN_DRIVER ?= native
+CAN_DRIVER ?= CAN_NATIVE
+
+# prevent using native driver on non-native board
+ifeq ($(CAN_DRIVER), CAN_NATIVE)
```

Comment:
Now that this is in place, the CI should only attempt to build for `native` right? Otherwise it would fail.

Reply:
Wouldn't it be possible to set the driver according to the used board and use the test for all of them?

Reply:
Mhmmm, yes. Because the Makefile will throw that error, the CI will fail indeed. Unless... the CI selects a can_driver when building for non-native. 
Could you guide me how I best solve this?

Reply:
Setting the driver according to the used board is not possible because because there is no correlation between board and driver. You can use any driver you want (as long as it's implemented in RIOT already) with any board you want.

Reply:
Taking a second look at the test, what is this actually testing when not running on native? For any board that is not native it will assert in the main function. Is it worth compiling this application for all boards in the CI? If not, the following can be added to the Makefile:

```Makefile
BOARD_WHITELIST := native
```

Reply:
> Is it worth compiling this application for all boards in the CI? If

To be honest I don't think it is useful, no. I'll add that line to the Makefile. It will not have any influence on making locally, right?

Reply:
>  It will not have any influence on making locally, right?

It would prevent local building as well, unless `CONTINUE_ON_EXPECTED_ERRORS=1` is set.

>(btw. I assume you meant adding it to Makefile.ci, not Makefile. Right?)

Actually I meant the `Makefile`, for some reason be place the list in that file also for other tests.

Reply:
> It would prevent local building as well, unless CONTINUE_ON_EXPECTED_ERRORS=1 is set.

Mhmmm, but... then why would we want to include this? 

> Actually I meant the Makefile, for some reason be place the list in that file also for other tests.

Oh, Ok. I will fix it once the above has been clarified.

Reply:
> Mhmmm, but... then why would we want to include this?

So the CI does not attempt to build the test for boards other than native.

Reply:
But you said this would also prevent users from building for non-native boards locally on their machines?

Reply:
> But you said this would also prevent users from building for non-native boards locally on their machines?

Yes, unless when compiling using `CONTINUE_ON_EXPECTED_ERRORS=1` (and of course overriding `CAN_DRIVER` so the check introduced here does not raise an error). What is the use case for running this test (as is) on a board that is not native?

Reply:
Are you suggesting that the user should always add `CONTINUE_ON_EXPECTED_ERRORS=1` himself? That doesn't feel like it's a good solution? I would expect it to work out of the box? 
User picks a board, picks a driver, and without any changes it works. 

> What is the use case for running this test (as is) on a board that is not native?

as is, not much. You will have to select a non-native driver. Either you implement the code for PERIPH_CAN (such as stm32 has), or you use the driver from the MCP2515 PR.

Reply:
> User picks a board, picks a driver, and without any changes it works.

Yes, I would expect that as well, which is not possible with the current state of the test, as it needs native implementation to work, or code modification in main to initialize the chosen driver. The whitelist can be removed once this limitation is not there anymore.

I don't really see the goal of this CFLAG when it is actually not pulling any driver module in...I would expect (when more drivers are there) to set this Makefile variable and that to be added to `USEMODULE`, like so:
https://github.com/RIOT-OS/RIOT/blob/7eed33bf7870a5e535d60f217acd70ac9cc05c0f/tests/driver_sx127x/Makefile#L10-L13

Reply:
Ok, I think I get what you're saying now. I added `BOARD_WHITELIST := native` to the makefile now.

> I don't really see the goal of this CFLAG when it is actually not pulling any driver module in
I removed it a few patches ago.